### PR TITLE
wp-env.json: Bump tt1-blocks dependency to v0.4.5

### PR DESCRIPTION
**Pull request by ockham**
## Description
As of #28741, we've been pinning the version of the Twenty Twenty One Blocks theme (`tt1-blocks`) used by Gutenberg to a fixed version, in order to prevent breakage of e2e tests by changes to that theme, as previously seen with https://github.com/WordPress/gutenberg/pull/28638.

The version we pinned `tt1-blocks` to was `0.4.3`. A new version -- ~`0.4.4`~ `0.4.5` -- has now been tagged, so let's try to update Gutenberg to use that.

## How has this been tested?
Verify that tests (unit and e2e) are passing (see cI). Some additional smoke testing (especially of the Site Editor) is also recommended.

### CODE REVIEW

**Review by kjellr — APPROVED** [review]
👍

**Comment by creativecoder** [comment]
Note there's at least 1 change that needs to be made to e2e tests with v0.4.4 of tt1-blocks: https://github.com/WordPress/gutenberg/blob/185574ecb7ec78a3de39587016c792a0e515ad9f/packages/e2e-tests/specs/experiments/template-part.test.js#L45

**Comment by ockham** [comment]
I just noticed that while the version of TT1 Blocks was indeed bumped to 0.4.4, no git tag was added (as e2e tests revealed:

```
GitError: fatal: couldn't find remote ref tt1-blocks@0.4.4
```

So it looks like we'll have to take care of that first 😬

**Comment by kjellr** [comment]
Sorry about that! I've [published a new release](https://github.com/WordPress/theme-experiments/releases/tag/tt1-blocks%400.4.4) for TT1 Blocks.

**Comment by ockham** [comment]
> Sorry about that! I've [published a new release](https://github.com/WordPress/theme-experiments/releases/tag/tt1-blocks%400.4.4) for TT1 Blocks.

Thanks a lot @kjellr!

**Comment by ockham** [comment]
> Note there's at least 1 change that needs to be made to e2e tests with v0.4.4 of tt1-blocks:
> 
> https://github.com/WordPress/gutenberg/blob/185574ecb7ec78a3de39587016c792a0e515ad9f/packages/e2e-tests/specs/experiments/template-part.test.js#L45

Thanks @creativecoder! Fixed in e7eef513aec1fefef0fa452a34df9fea0e6dfb39.

**Comment by ockham** [comment]
e2e tests are currently failing at the following line:

https://github.com/WordPress/gutenberg/blob/72955796a533bdbc198a82a35074c0a0479b63c0/packages/e2e-tests/specs/experiments/settings-sidebar.test.js#L90-L91

It seems like the `getAllBlocks()` call now returns `undefined` 😕  I didn't see at a glance what might've changed in TT1-Blocks v0.4.4.

**Comment by ockham** [comment]
Sorry for the noise, I botched a rebase while using the GH Desktop app. New computer! 😅

**Comment by david-szabo97** [comment]
Few more

https://github.com/WordPress/gutenberg/blob/2211285672ae5131d49ff61d3e77773848b959a7/packages/e2e-tests/specs/experiments/multi-entity-editing.test.js#L152

https://github.com/WordPress/gutenberg/blob/f8a84651003a29c2dd469e7537fce17f42b4cd6a/packages/e2e-tests/specs/experiments/document-settings.test.js#L80

**Comment by ockham** [comment]
> Few more
> 
> https://github.com/WordPress/gutenberg/blob/2211285672ae5131d49ff61d3e77773848b959a7/packages/e2e-tests/specs/experiments/multi-entity-editing.test.js#L152
> 
> https://github.com/WordPress/gutenberg/blob/f8a84651003a29c2dd469e7537fce17f42b4cd6a/packages/e2e-tests/specs/experiments/document-settings.test.js#L80

Thanks @david-szabo97, I'll fix those! BTW would you have any idea where [this error](https://github.com/WordPress/gutenberg/pull/30045#issuecomment-805233742) is coming from? (I saw that you originally wrote that test 😬 )

**Comment by david-szabo97** [comment]
> BTW would you have any idea where this error is coming from? (I saw that you originally wrote that test 😬 )

This is really strange. Theme shouldn't affect this at all. `getAllBlocks` runs `window.wp.data.select('core/block-editor').getBlocks()` on the page.

**Comment by ockham** [comment]
> > BTW would you have any idea where this error is coming from? (I saw that you originally wrote that test 😬 )
> 
> This is really strange. Theme shouldn't affect this at all. `getAllBlocks` runs `window.wp.data.select('core/block-editor').getBlocks()` on the page.

Yeah, it's quite weird, isn't it 😕  I had a hunch that it might have to do with the iframing of the site editor (and `window.wp` being unvailable from the context it's called from), but that would be purely on the Gutenberg side, and totally unrelated to the theme itself, wouldn't it 🤔

**Comment by Addison-Stavlo** [comment]
I notice that here we updated the JS tests for template parts in the sidebar to now be 'Header' as opposed to 'General'.

I would have expected PHP unit to fail with these same theme updates without updating this as well 🤔 :
https://github.com/WordPress/gutenberg/blob/df8e7307404818446c3ffd0d86ac6584fa15a911/phpunit/class-block-templates-test.php#L205-L206
It seems to still be passing though.

**Comment by ockham** [comment]
> I notice that here we updated the JS tests for template parts in the sidebar to now be 'Header' as opposed to 'General'.
> 
> I would have expected PHP unit to fail with these same theme updates without updating this as well 🤔 :
> https://github.com/WordPress/gutenberg/blob/df8e7307404818446c3ffd0d86ac6584fa15a911/phpunit/class-block-templates-test.php#L205-L206
> 
> 
> It seems to still be passing though.

Ah, interesting indeed 🤔  Is it possible that both `WP_TEMPLATE_PART_AREA_UNCATEGORIZED` and `$template->area` are undefined here? I've noticed that `WP_TEMPLATE_PART_AREA_UNCATEGORIZED` is defined here: https://github.com/WordPress/gutenberg/blob/e7369c09f468158e42851ba2a5b1b58efbd4ca88/lib/full-site-editing/template-parts.php#L102-L104

Any chance that `template-parts.php` isn't loaded on some code paths?

**Comment by kjellr** [comment]
Just a heads up that we've bundled [a new 0.4.5 version of TT1 Blocks](https://github.com/WordPress/theme-experiments/releases/tag/tt1-blocks%400.4.5) to coincide with Gutenberg 10.3's release today.

**Comment by ockham** [comment]
> I notice that here we updated the JS tests for template parts in the sidebar to now be 'Header' as opposed to 'General'.
> 
> I would have expected PHP unit to fail with these same theme updates without updating this as well 🤔 :
> https://github.com/WordPress/gutenberg/blob/df8e7307404818446c3ffd0d86ac6584fa15a911/phpunit/class-block-templates-test.php#L205-L206
> 
> 
> It seems to still be passing though.

FWIW, I tried updating those in 5eb999b34adbad7f6cfbcb444056d640022593c8, but that caused PHP unit tests to fail, so I had to revert.

**Comment by ockham** [comment]
e2e tests are passing now 🎉

**Comment by Addison-Stavlo** [comment]
> Is it possible that both WP_TEMPLATE_PART_AREA_UNCATEGORIZED and $template->area are undefined here?

I don't think so, just testing it out they are outputting the expected string values when the tests run ('header','uncategorized',etc.) and not `undefined`.  Do php unit tests on CI somehow run off the theme version on `trunk` and not the PR or something? 🤔   Anyways, I don't mean to stall this PR on it but its something to be aware of in case they start failing after merge or something.

**Comment by ockham** [comment]
> > Is it possible that both WP_TEMPLATE_PART_AREA_UNCATEGORIZED and $template->area are undefined here?
> 
> I don't think so, just testing it out they are outputting the expected string values when the tests run ('header','uncategorized',etc.) and not `undefined`. Do php unit tests on CI somehow run off the theme version on `trunk` and not the PR or something? 🤔

Doesn't seem like: https://github.com/WordPress/gutenberg/blob/2557a44f182041394cd9bce173ee8355849418d7/.github/workflows/unit-test.yml#L95-L101

Plus I can repro locally.

I checked the locally generated `docker-compose.yml`, it seems fine -- volumes for TT1 Blocks are mounted for pretty much all relevant services (`wordpress`, `tests-wordpress`, `phpunit`) 🤔 

TT1 Blocks seems to set template areas fine: https://github.com/WordPress/theme-experiments/blob/tt1-blocks%400.4.5/tt1-blocks/experimental-theme.json

> Anyways, I don't mean to stall this PR on it but its something to be aware of in case they start failing after merge or something.

Yeah, definitely something we should figure out 🤔 

Any chance there's actually something wrong with ~the controller logic~ _edit:_ `gutenberg_get_block_template`?

**Comment by ockham** [comment]
Using this to track down the issue:

```
npm run test-unit-php -- /var/www/html/wp-content/plugins/gutenberg/phpunit/class-block-templates-test.php
```

**Comment by ockham** [comment]
I believe I've tracked it down to this line returning `false`: https://github.com/WordPress/gutenberg/blob/aa05bd47df3247730065b197a170ab84e4c6761f/lib/full-site-editing/block-templates.php#L127

**Comment by ockham** [comment]
Tracked it down further: `WP_Theme_JSON_Resolver::theme_has_support()` tries to locate `experimental-theme.json` in `get_stylesheet_directory()`:

https://github.com/WordPress/gutenberg/blob/2557a44f182041394cd9bce173ee8355849418d7/lib/class-wp-theme-json-resolver.php#L513-L538

But when this function is called, `get_stylesheet_directory()` points to the default theme's directory for me.

So I guess it's another instance of the "calling a theme related function before WP has switched to that theme" 😬 

cc/ @nosolosw

**Comment by ockham** [comment]
> Tracked it down further: `WP_Theme_JSON_Resolver::theme_has_support()` tries to locate `experimental-theme.json` in `get_stylesheet_directory()`:
> [...]
> But when this function is called, `get_stylesheet_directory()` points to the default theme's directory for me.
> 
> So I guess it's another instance of the "calling a theme related function before WP has switched to that theme" 😬

To be precise, I think that that happens when `WP_Theme_JSON_Resolver::theme_has_support()` is called for the first time (possibly before _this_ logic is run?) But that sets `WP_Theme_JSON_Resolver::$theme_has_support` to `false`, and that cache value is then used as the return value for `WP_Theme_JSON_Resolver::theme_has_support()` the next time it's called.

**Comment by ockham** [comment]
> > Tracked it down further: `WP_Theme_JSON_Resolver::theme_has_support()` tries to locate `experimental-theme.json` in `get_stylesheet_directory()`:
> > [...]
> > But when this function is called, `get_stylesheet_directory()` points to the default theme's directory for me.
> > So I guess it's another instance of the "calling a theme related function before WP has switched to that theme" 😬
> 
> To be precise, I think that that happens when `WP_Theme_JSON_Resolver::theme_has_support()` is called for the first time (possibly before _this_ logic is run?) But that sets `WP_Theme_JSON_Resolver::$theme_has_support` to `false`, and that cache value is then used as the return value for `WP_Theme_JSON_Resolver::theme_has_support()` the next time it's called.

Filed an issue: https://github.com/WordPress/gutenberg/issues/30478

**Comment by ockham** [comment]
I've unreverted the unit tests update, which will cause PHP unit tests to fail on this PR. This will block the PR until #30478 is fixed (hopefully by #30479), which I think is okay.

**Comment by oandregal** [comment]
:wave: Not sure if relevant, but I see this notice message when loading a request:

```
[05-Apr-2021 10:32:34 UTC] PHP Notice:  Undefined property: WP_Block_Template::$area in /home/andres/src/gutenberg/lib/full-site-editing/class-wp-rest-templates-controller.php on line 413
[05-Apr-2021 10:32:34 UTC] PHP Stack trace:
[05-Apr-2021 10:32:34 UTC] PHP   1. {main}() /home/andres/flywheel-sites/gutenbergdev/app/public/index.php:0
[05-Apr-2021 10:32:34 UTC] PHP   2. require() /home/andres/flywheel-sites/gutenbergdev/app/public/index.php:17
[05-Apr-2021 10:32:34 UTC] PHP   3. wp() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-blog-header.php:16
[05-Apr-2021 10:32:34 UTC] PHP   4. WP->main() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-includes/functions.php:1291
[05-Apr-2021 10:32:34 UTC] PHP   5. WP->parse_request() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-includes/class-wp.php:750
[05-Apr-2021 10:32:34 UTC] PHP   6. do_action_ref_array() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-includes/class-wp.php:388
[05-Apr-2021 10:32:34 UTC] PHP   7. WP_Hook->do_action() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-includes/plugin.php:551
[05-Apr-2021 10:32:34 UTC] PHP   8. WP_Hook->apply_filters() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-includes/class-wp-hook.php:316
[05-Apr-2021 10:32:34 UTC] PHP   9. rest_api_loaded() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-includes/class-wp-hook.php:292
[05-Apr-2021 10:32:34 UTC] PHP  10. WP_REST_Server->serve_request() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-includes/rest-api.php:354
[05-Apr-2021 10:32:34 UTC] PHP  11. WP_REST_Server->dispatch() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-includes/rest-api/class-wp-rest-server.php:412
[05-Apr-2021 10:32:34 UTC] PHP  12. WP_REST_Server->respond_to_request() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-includes/rest-api/class-wp-rest-server.php:985
[05-Apr-2021 10:32:34 UTC] PHP  13. WP_REST_Templates_Controller->get_items() /home/andres/flywheel-sites/gutenbergdev/app/public/wp-includes/rest-api/class-wp-rest-server.php:1139
[05-Apr-2021 10:32:34 UTC] PHP  14. WP_REST_Templates_Controller->prepare_item_for_response() /home/andres/src/gutenberg/lib/full-site-editing/class-wp-rest-templates-controller.php:146
```

(leaving a comment on the issue about theme.json support https://github.com/WordPress/gutenberg/issues/30478, not sure I can repro the failure)

**Comment by Addison-Stavlo** [comment]
> Filed an issue: #30478

Great digging, thanks for figuring out the cause of that!  I think what you are proposing on that issue makes sense.

**Comment by ockham** [comment]
The issue should now be fixed by #30830. Going to rebase 🤞

**Comment by ockham** [comment]
Pushed one more commit to fix one remaining unit test. Should be ready for another look now 🙂

**Pull request merged**
